napari/_commands/_layers.py

```diff
@@ -0,0 +1,248 @@
+"""LayerCommands class and helpers.
+"""
+from typing import List
+
+import dask.array as da
+
+from ._humanize import naturalsize
+from ._tables import print_property_table, RowTable
+from ._utils import highlight
+from ..layers.base import Layer
+
+
+HELP_STR = f"""
+{highlight("Available Commands:")}
+cmd.help
+cmd.list
+cmd.info(layer_id)
+"""
+
+
+def _get_type_str(data) -> str:
+    """Get human readable name for the data's type.
+
+    Returns
+    -------
+    str
+        A string like "ndarray" or "dask".
+    """
+    if isinstance(data, list):
+        if len(data) == 0:
+            return "EMPTY"
+        else:
+            # Recursively get the type string of the zeroth level.
+            return _get_type_str(data[0])
+
+    if type(data) == da.Array:
+        # Special case this because otherwise data_type.__name__
+        # below would just return "Array".
+        return "dask"
+
+    # For class numpy.ndarray this returns "ndarray"
+    return type(data).__name__
+
+
+def _get_size_str(data) -> str:
+    """Return human readable size.
+
+    Parameters
+    ----------
+    data
+        Layer's data.
+
+    Returns
+    -------
+    str
+        A string size like "24.2G".
+    """
+    if isinstance(data, list):
+        nbytes = sum(level.nbytes for level in data)
+    else:
+        nbytes = data.nbytes
+    return naturalsize(nbytes, gnu=True)
+
+
+class ListLayersTable:
+    """Table showing information about each layer.
+
+    Parameters
+    ----------
+    layers : List[Layer]
+        The layers to list in the table.
+
+    Attributes
+    ----------
+    table : TextTable
+        Formats our table for printing.
+    """
+
+    def __init__(self, layers: List[Layer]):
+        self.layers = layers
+        self.table = RowTable(
+            [
+                "ID",
+                {"name": "NAME", "align": "left"},
+                "LAYER",
+                "DATA",
+                "LEVELS",
+                "SHAPE",
+                "TOTAL",
+            ]
+        )
+        for i, layer in enumerate(self.layers):
+            self._add_row(i, layer)
+
+    def _get_shape_str(self, data):
+        """Get shape string for the data.
+
+        Either "NONE" or a tuple like "(10, 100, 100)".
+        """
+        if isinstance(data, list):
+            if len(data) == 0:
+                return "NONE"  # Shape layer is empty list?
+            else:
+                return f"{data[0].shape}"  # Multi-scale
```

Comment:
This will also return the shape of the first `Shape` in a shapes layer. I guess I'd rather return the number of shapes (i.e. length of the list in this case).

We actually have a `layer.shape` property, but it is being deprecated/ not what you want to use. We could change it now though so it was what you wanted to use